tservice/src/seal.rs

```diff
@@ -0,0 +1,112 @@
+// Copyright (c) 2022 The MobileCoin Foundation
+//! Functions used for sealing and unsealing of secrets
+
+use alloc::{vec, vec::Vec};
+pub use core::ptr;
+pub use mc_sgx_core_types::{Error, Result};
+pub use mc_sgx_tservice_sys::sgx_calc_sealed_data_size;
+pub use mc_sgx_tservice_sys_types::sgx_sealed_data_t;
+pub use mc_sgx_tservice_types::Sealed;
+use mc_sgx_util::ResultInto;
+
+/// Sealed data builder
+#[derive(Debug, Clone, Hash, PartialEq, Eq, Default)]
+pub struct SealedBuilder<T> {
+    /// The data to be encrypted/sealed
+    data: T,
+
+    /// The AAD(additional authenticated data) to use in the sealing.
+    /// The Intel docs often refer to this as _MAC text_
+    aad: Option<T>,
+}
+
+impl<T: AsRef<[u8]> + core::default::Default> SealedBuilder<T> {
+    /// A [`SealedBuilder`] from
+    ///
+    /// # Arguments
+    /// * `data` - The data to be encrypted/sealed
+    pub fn new(data: T) -> Self {
+        Self {
+            data,
+            ..Default::default()
+        }
+    }
+
+    /// Build the [`Sealed`] object
+    pub fn build(self) -> Result<Sealed<Vec<u8>>> {
```

Comment:
I think this could probably be non-consuming (`&self` instead of `self`), as well as the other builder methods (`&mut self` instead of `mut self`), since the `sgx_seal_data` call doesn't take ownership of `self.data` or `self.aad`. [The Rust style guidelines recommend using non-consuming builders where possible.](https://doc.rust-lang.org/1.0.0/style/ownership/builders.html)

Reply:
nice find on the builder guidance, will work to update as suggested
Thanks!